jax/core.py

```diff
@@ -123,8 +136,12 @@ def jaxpr_as_fun(typed_jaxpr: TypedJaxpr, *args):
 
 
 
-class JaxprEqn(namedtuple('JaxprEqn',
-                          ['invars', 'outvars', 'primitive', 'params'])):
+class JaxprEqn(NamedTuple):
+  invars: List[Atom]
+  outvars: List['Var']
+  primitive: 'Primitive'
+  params: Dict[str, Any]
```

Comment:
Not sure if there is a type for this, but I think we require parameters to be hashable, like `Dict[str, Hashable]` in principle at least.

Reply:
Done, although it required adding a couple more `casts` when there is a protocol that mypy can't understand (e.g., `jaxpr` entries are `Jaxpr` typed.)